File: project-euler/86.py


from __future__ import print_function
from math import sqrt
from collections import Counter
from pprint import pprint


def main():
    count = 0
    a = 0
    while True:
        a += 1
        for b in range(1, a + 1):
            for c in range(1, b + 1):
                shortest = sqrt(a * a + (b + c) * (b + c))
                if abs(shortest - int(shortest)) < 1e-6:
                    print("a: %s, b: %s, c: %s :: shortest: %s" % (a, b, c, shortest))
                    count += 1
                    if count >= 100:
                        print(a)
                        return
    print(count)


if __name__ == "__main__":
    main()

File: project-euler/86.jl

function main()
    count = 0
    a = 0
    while true
        a += 1
        for b=1:a, c=1:b
            shortest = sqrt(a * a + (b + c) * (b + c))
            if abs(shortest - round(Int, shortest)) < 1e-6
                count += 1
                if count > 1000000
                    println(a)
                    return
                end
            end
        end
    end
end


main()
